src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -285,4 +286,132 @@ function (int $shopId, array $productImages): ShopProductImages {
+            ->setParameter('cover', $image->cover ? 1 : null)
+            ->execute()
+        ;
+    }
+
+    public function updateMissingCovers(ProductId $productId): void
+    {
+        $results = $this->connection->createQueryBuilder()
+            ->select('is.id_image', 'is.id_shop', 'is.cover')
+            ->from($this->dbPrefix . 'image_shop', '`is`')
+            ->leftJoin(
+                '`is`',
+                $this->dbPrefix . 'image',
+                'i',
+                'i.id_image = is.id_image'
+            )
+            ->andWhere('is.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->addOrderBy('i.position', 'ASC')
+            ->execute()
+            ->fetchAll()
+        ;
+
+        foreach ($results as $image) {
+            $coverId = $this->getCoverImageId($productId, new ShopId((int) $image['id_shop']));
+            if ($coverId !== null && $coverId->getValue() === (int) $image['id_image']) {
+                continue;
+            }
+
+            if ($coverId === null) {
+                $newValue = 1;
+            } else {
+                $newValue = null;
+            }
+
+            if ($newValue === $image['cover']) {
+                continue;
+            }
+            $this->connection->createQueryBuilder()
+                ->update($this->dbPrefix . 'image_shop')
+                ->set($this->dbPrefix . 'image_shop' . '.cover', ':cover')
+                ->setParameter('cover', $newValue)
+                ->andWhere($this->dbPrefix . 'image_shop' . '.id_image = :imageId')
+                ->setParameter('imageId', (int) $image['id_image'])
+                ->andWhere($this->dbPrefix . 'image_shop' . '.id_shop = :shopId')
+                ->setParameter('shopId', (int) $image['id_shop'])
+                ->execute()
+            ;
+        }
+    }
+
+    /**
+     * @param array<int|string, string|int[]> $updatableProperties
+     * @param ShopId[] $shopIds
+     */
+    public function partialUpdateForShops(Image $image, array $updatableProperties, array $shopIds, int $errorCode = 0): void
+    {
+        $this->productImageValidator->validate($image);
+        $this->partiallyUpdateObjectModelForShops(
+            $image,
+            $updatableProperties,
+            $shopIds,
+            CannotUpdateProductImageException::class,
+            $errorCode
+        );
+    }
+
+    public function delete(Image $image): void
+    {
+        $this->deleteObjectModelFromShops(
+            $image,
+            $this->getAssociatedShopIds(new ImageId((int) $image->id)),
+            CannotDeleteProductImageException::class
+        );
+    }
+
+    /**
+     * @param ImageId $imageId
+     *
+     * @return Image
+     *
+     * @throws CoreException
+     */
+    public function getImageForDefaultShop(ImageId $imageId): Image
+    {
+        /** @var Image $image */
+        $image = $this->getObjectModel(
```

Comment:
Technically it would work for your use case, but it doesn't really get the one from default shop because it will rely on the internet ObjectModel fallback on the context, and we don't want to rely on it anymore

So you'll need to get the real associated default shop ID for the product. Two possibilities for this:
- get the ProductRepository and use the `getProductDefaultShopId` method to get the proper ShopID, but it means you'll have to get the product ID based on ImageId first so maybe it's not the best solution (see solution2)
- create a private method to get back the default shopID by doing a leftJoin from `image` table to `product` table